lib/_stream_readable.js

```diff
@@ -152,6 +154,28 @@ Readable.prototype.isPaused = function() {
   return this._readableState.flowing === false;
 };
 
+Readable.prototype.destroy = function() {
+  var self = this;
+  if (this._readableState.destroyed) return;
+  this._readableState.destroyed = true;
+
+  // if this is a duplex stream mark the writable part as destroyed as well
+  if (this._writableState) {
+    this._writableState.destroyed = true;
+  }
```

Comment:
Sort of tempted to say "may as well specially implement `destroy` on Duplex" just to avoid adding another case of the parent class having intimate knowledge of its subclass.